servers/rendering/shader_preprocessor.cpp

```diff
@@ -368,6 +368,7 @@ String ShaderPreprocessor::vector_to_string(const LocalVector<char32_t> &p_v, in
 
 String ShaderPreprocessor::tokens_to_string(const LocalVector<Token> &p_tokens) {
 	LocalVector<char32_t> result;
+	result.reserve(p_tokens.size());
```

Comment:
This may be better served with an in-place `String` construction; though I'm not sure whether the `token.text` can be trusted to be correct `UTF-32`.